components.js

```diff
@@ -1,257 +1 @@
-
- /**
- * TextInput component.
- *
- * @type {import('react').ComponentType<{ hint: string, name: string, label: string, fieldError: string, multiline: boolean, description: string }>}
- *
- * @example
- * 
- * import { TextInput } from 'camunda-modeler-plugin-helpers/components';
- *
- * function CustomInput(props) {
- *   return (
- *    <Formik initialValues={ initialValues } onSubmit={ this.onSubmit }>
- *      {() => (
- *        <Form>
- *          <Field
- *            component={ TextInput }
- *            label="My input"
- *            id={ id }
- *            multiline={ false }
- *            name={ name }
- *            description="Custom description"
- *          />
- *        </Form>
- *       )}
- *    </Formik>
- *   );
- * }
- */
-export const TextInput = window.components.TextInput || NotCompatible('5.29');
-
- /**
- * CachedComponent class.
- *
- * @type {import('react').ComponentClass}
- *
- * @example
- * 
- * import { CachedComponent } from 'camunda-modeler-plugin-helpers/components';
- * 
- * class ComponentWithCachedState extends CachedComponent {
- *  constructor(props) {
- *   super(props);
- *  }
- * 
- *  getCachedState() {
- *    return this.getCached()
- *  }
- * 
- *  setCachedState(values) {
- *    this.setCached(values)
- *  }
- * }
- * 
- */
-export const CachedComponent = window.components.CachedComponent || NotCompatible('5.29');
-
-/**
- * A higher order component that passes cache to a wrapped component.
- * Forwards refs, too.
- * 
- * @type {Function}
- * @param {Component} Comp
- */
-export const WithCache = window.components.WithCache || NotCompatible('5.29');
-
-/**
- * A higher order component that lazily
- * initiates the given wrapped component
- * via the `Comp#createCachedState` method.
- *
- * Passes props as well as destructured
- * wrapped component state to `Comp`.
- *
- * The resulting component must be called
- * with the `id` and `cache` prop.
- *
- * Forwards refs, too.
- *
- * @type {Function}
- * @param {Component} Comp
- */
-export const WithCachedState = window.components.WithCachedState || NotCompatible('5.29');
-
-/**
- * A helper function to create Tab components
- * to be used with the TabProvider.
- *
- * @type {Function}
- * @param {string} tabName - The name of the tab.
- * @param {object} providers - The providers object.
- * @param {string} providers.type - The type of the provider.
- * @param {React.Component} providers.editor - The editor component.
- * @param {string} providers.defaultName - The default name of the provider.
- * @returns {React.Component} The created EditorTab component.
- */
-export const createTab = window.components.createTab || NotCompatible('5.29');
\ No newline at end of file
+throw new Error('camunda-modeler-plugin-helpers/components.js is no longer provided. Migrate to individual component exports: https://github.com/camunda/camunda-modeler-plugin-helpers/pull/31');
```

Comment:
Shift left! I like it.

Reply:
But shift left would be:

* Remove the exports, so this fails during build time already.

Reply:
As per "further shift left" I removed https://github.com/camunda/camunda-modeler-plugin-helpers/commit/13fe3dfec7e11665e00449be9fc5a6bbc884fd6e again.

Reply:
Good solution!